src/server/graph-generation/treeTraversal.js

```diff
@@ -0,0 +1,114 @@
+/**
+    Pathway Commons Central Data Cache
+
+    Tree Traversal
+    treeTraversal.js
+
+    Purpose  : To provide functions to search and traverse a tree represented using 
+               javascript arrays
+
+    Requires : Valid tree array
+
+    @author Harsh Mistry
+    @version 1.1 2017/10/17
+**/
+
+//Returns all values mapped to a key in a given subtree
+//Requires subtree to be valid
+//Note : - A empty array is returned if no match is found
+//       - Recurse indicates if other levels of the tree should be searched
+function searchTree(subtree, key, recurse = true) {
+    var result = []
+  
+    //Loop through all level nodes
+    for (var i = 0; i < subtree.length; i++) {
+      //Push current value if it matches
+      if (subtree[i][0] == key) {
+        result.push(subtree[i][1]);
+      }
+  
+      //Recurse on subtree if one exists
+      if (subtree[i][1] instanceof Array && recurse) {
+        result.push(searchTree(subtree[i][1], key));
+      }
+    }
+  
+    //Remove all invalid values
+    for (var i = 0; i < result.length; i++) {
+      if (!(result[i])) {
+        result.splice(i, 1);
```

Comment:
Again, reverse order or mind your indices

Reply:
Or just use `_.compact()` to remove falsey values : https://lodash.com/docs#compact

Reply:
Parser